text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,678 @@
+- The target must provide documentation for the Rust community explaining how
+  to build for the target, using cross-compilation if possible. If the target
+  supports running tests (even if they do not pass), the documentation must
+  explain how to run tests for the target, using emulation if possible or
+  dedicated hardware if necessary.
+- Tier 3 targets must not impose burden on the authors of pull requests, or
+  other developers in the community, to maintain the target. In particular,
+  do not post comments (automated or manual) on a PR that suggests a block on
+  the PR based on the target. (A PR author may choose to help with a tier 3
+  target, but is not required to.)
+  - It is acceptable to manually initiate or link to a discussion about how to
+    make a PR and a tier 3 target compatible, but that discussion must not be
+    implied to block the PR unless either the PR author or PR approver want it
+    to be.
+- Patches adding or updating tier 3 targets must not break any existing tier 2
+  or tier 1 target, and must not break another tier 3 target without approval
+  of either the compiler team or the maintainers of the other tier 3 target.
+  - In particular, this may come up when working on closely related targets,
+    such as variations of the same architecture with different features. Avoid
+    introducing unconditional uses of features that another variation of the
+    target may not have; use conditional compilation or runtime detection, as
+    appropriate, to let each target run code supported by that target.
+
+If a tier 3 target stops meeting these requirements, or the target maintainers
+no longer have interest or time, or the target shows no signs of activity and
+has not built for some time, or removing the target would improve the quality
+of the Rust codebase, we may post a PR to remove it; any such PR will be CCed
+to the target maintainers (and potentially other people who have previously
+worked on the target), to check potential interest in improving the situation.
+
+## Tier 2 target policy
+
+At this tier, the Rust project guarantees that a target builds, and will reject
+patches that fail to build on a target. Thus, we place requirements that ensure
+the target will not block forward progress of the Rust project.
+
+A proposed new tier 2 target must be reviewed and approved by the compiler team
+based on these requirements. Such review and approval may occur via a Major
+Change Proposal (MCP).
+
+In addition, the infrastructure team must approve the integration of the target
+into Continuous Integration (CI), and the tier 2 CI-related requirements. This
+review and approval may take place in a PR adding the target to CI, or simply
+by an infrastructure team member reporting the outcome of a team discussion.
+
+- A tier 2 target must have value to people other than its maintainers. (It may
+  still be a niche target, but it must not be exclusively useful for an
+  inherently closed group.)
```

Comment:
I am interested in why this requirement is only present for tier 2 targets, rather than for tier 3 targets as well.

Reply:
I intentionally added this requirement just to tier 2 targets, because I think it's acceptable for a tier 3 target to *initially* only have value to the set of people who write and maintain it. Tier 3 is the "we promise nothing" tier, but we still require that teams approve tier 3 targets, so the approvers must see *some* value and future possibilities in the target. Having this requirement at tier 2 but not tier 3 allows for situations like pre-release hardware, test groups, not-yet-documented platforms, or other steps that may enable the interest and inertia needed to subsequently meet all the tier 2 requirements, including this one.